engine/pkg/meta/internal/sqlkv/sql_impl.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	db, err := gorm.Open(mysql.New(mysql.Config{
+		Conn:                      sqlDB,
+		SkipInitializeWithVersion: false,
+	}), &gorm.Config{
+		SkipDefaultTransaction: true,
+		// TODO: logger
+	})
+	if err != nil {
+		return nil, errors.ErrMetaNewClientFail.Wrap(err)
+	}
+
+	tableScopeDB := db
+	if table != "" {
+		tableScopeDB = db.Table(table)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	impl := &sqlImpl{
+		db:           db,
+		jobID:        jobID,
+		tableScopeDB: tableScopeDB,
+		epochClient:  ormModel.NewEpochClient(jobID, db),
+	}
+	if err := impl.InitializeModel(ctx); err != nil {
+		return nil, err
+	}
+
+	return impl, nil
+}
+
+// InitializeModel initializes metakv table
+func (c *sqlImpl) InitializeModel(ctx context.Context) error {
+	if err := c.tableScopeDB.
+		WithContext(ctx).
+		AutoMigrate(&sqlkvModel.MetaKV{}); err != nil {
+		return cerrors.ErrMetaOpFail.Wrap(err)
+	}
+
+	return c.epochClient.Initialize(ctx)
+}
+
+// Close implements Close interface of Client
+func (c *sqlImpl) Close() error {
+	return nil
+}
+
+// GetEpoch implements GenEpoch interface of Client
+func (c *sqlImpl) GenEpoch(ctx context.Context) (int64, error) {
+	return c.epochClient.GenEpoch(ctx)
+}
+
+// Put implements Put interface of KV
+func (c *sqlImpl) Put(ctx context.Context, key, val string) (*metaModel.PutResponse, metaModel.Error) {
+	op := metaModel.OpPut(key, val)
+	return c.doPut(ctx, c.tableScopeDB, &op)
+}
+
+func (c *sqlImpl) doPut(ctx context.Context, db *gorm.DB, op *metaModel.Op) (*metaModel.PutResponse, metaModel.Error) {
+	if err := db.WithContext(ctx).
+		Clauses(clause.OnConflict{
+			UpdateAll: true,
+		}).Create(&sqlkvModel.MetaKV{
+		JobID: c.jobID,
+		KeyValue: metaModel.KeyValue{
+			Key:   op.KeyBytes(),
+			Value: op.ValueBytes(),
+		},
+	}).Error; err != nil {
+		return nil, sqlErrorFromOpFail(err)
+	}
+
+	return &metaModel.PutResponse{
+		Header: &metaModel.ResponseHeader{},
+	}, nil
+}
+
+// Get implements Get interface of KV
+func (c *sqlImpl) Get(ctx context.Context, key string, opts ...metaModel.OpOption) (*metaModel.GetResponse, metaModel.Error) {
+	op := metaModel.OpGet(key, opts...)
+	return c.doGet(ctx, c.tableScopeDB, &op)
+}
+
+func (c *sqlImpl) doGet(ctx context.Context, db *gorm.DB, op *metaModel.Op) (*metaModel.GetResponse, metaModel.Error) {
+	if err := op.CheckValidOp(); err != nil {
+		return nil, &sqlError{
+			displayed: cerrors.ErrMetaOptionInvalid.Wrap(err),
+		}
+	}
+
+	var (
+		metaKvs    []*sqlkvModel.MetaKV
+		metaKv     sqlkvModel.MetaKV
+		err        error
+		isPointGet bool
+		key        = op.KeyBytes()
+	)
+
+	db = db.WithContext(ctx).Where("job_id = ?", c.jobID)
+	switch {
+	case op.IsOptsWithRange():
+		err = db.Where("key >= ? AND key < ?", key, op.RangeBytes()).Find(&metaKvs).Error
+	case op.IsOptsWithPrefix():
+		err = db.Where("key like ?%", key).Find(&metaKvs).Error
```

Comment:
I'm not sure whether it works for non-ascii key. Could you please add a test for it?